javalib/src/main/scala/java/lang/System.scala

```diff
@@ -132,50 +135,4 @@ object System {
 
     Collections.unmodifiableMap(map)
   }
-
-  private class CFileOutputStream(stream: Ptr[stdio.FILE])
-      extends java.io.OutputStream {
-    private val buf = stdlib.malloc(1)
-    def write(b: Int): Unit = {
-      !buf = b.toUByte.toByte
-      stdio.fwrite(buf, 1, 1, stream)
-    }
-  }
-
-  private class CStdinStream() extends java.io.InputStream {
-    private def readAndRetry(fd: CInt,
-                             buf: Ptr[scala.Byte],
-                             count: CSize): CSize = {
-      var nread: CSize = -1
-      do {
-        nread = unistd.read(unistd.STDIN_FILENO, buf, count)
-        if (nread == -1 && errno.errno != EINTR)
-          throw new IOException("Error on reading stdin")
-      } while (nread == -1)
-      nread
-    }
-
-    def read(): CInt = {
-      val buffer = stackalloc[scala.Byte](1)
-      var nread  = readAndRetry(unistd.STDIN_FILENO, buffer, 1)
```

Comment:
I suggest to port the error checking and retry logic to FileInputStream. Right now it simply ignore errors.